icu4c/source/i18n/number_output.cpp

```diff
@@ -39,6 +39,48 @@ MeasureUnit FormattedNumber::getOutputUnit(UErrorCode& status) const {
     return fData->outputUnit;
 }
 
+NounClass FormattedNumber::getNounClass(UErrorCode &status) const {
+    UPRV_FORMATTED_VALUE_METHOD_GUARD(NounClass::NEUTER);
+    const char *nounClass = fData->gender;
+
+    // if it is not exist, return `NEUTER`
+    if (uprv_strcmp(nounClass, "") == 0) {
+        return NounClass::NEUTER;
+    }
+
+    if (uprv_strcmp(nounClass, "neuter") == 0) {
+        return NounClass::NEUTER;
+    }
+
+    if (uprv_strcmp(nounClass, "feminine") == 0) {
+        return NounClass::FEMININE;
+    }
+
+    if (uprv_strcmp(nounClass, "masculine") == 0) {
+        return NounClass::MASCULINE;
+    }
+
+    if (uprv_strcmp(nounClass, "animate") == 0) {
+        return NounClass::ANIMATE;
+    }
+
+    if (uprv_strcmp(nounClass, "inanimate") == 0) {
+        return NounClass::INANIMATE;
+    }
+
+    if (uprv_strcmp(nounClass, "personal") == 0) {
+        return NounClass::PERSONAL;
+    }
+
+    if (uprv_strcmp(nounClass, "common") == 0) {
+        return NounClass::COMMON;
+    }
+
+    // In case there is no matching
+    status = U_INTERNAL_PROGRAM_ERROR;
+    return NounClass::NEUTER;
+}
+
 const char *FormattedNumber::getGender(UErrorCode &status) const {
```

Comment:
Better to remove it next cycle so that people can migrate more easily.

Reply:
agreed then :)

Reply:
shall I changed in this pull request, or make in a separate one?

Reply:
Okay, I will make it in a different PR.

Reply:
https://unicode-org.atlassian.net/browse/ICU-21925

Reply:
You should change the `@internal ICU 69 technology preview` to `@deprecated ICU 70` in this PR because it affects the public API. We want to stabilize the public API.

Reply:
we just need to make it `@deprecated` without specifying a version.